bin/csv_from_oai.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+require 'oai'
+require_relative "../lib/oai/client_decorator"
+
+# Input for dynamic email
+puts "Enter your email address:"
+email = gets.chomp
+
+# OAI client setup
+client = OAI::Client.new(
+  "http://oai.adventistdigitallibrary.org/OAI-script",
+  headers: { from: email },
+  parser: 'libxml'
+)
+
+opts = {
+  metadata_prefix: "oai_adl"
+}
+
+CSV.open('csv_from_oai.csv', 'wb') do |csv|
+  # Write the headers to the CSV file
+  csv << ['set', 'identifier', 'derivative_type', 'URL']
+
+  sets = [
+    "adl:thesis",
+    "adl:periodical",
+    "adl:issue",
+    "adl:article",
+    "adl:image",
+    "adl:book",
+    "adl:other"
+  ]
+
+  sets.each do |set|
+    records = client.list_records(opts.merge(set: set))
```

Comment:
Let's default to the `records.full.each_with_index`; that's the data we're after.

Reply:
Consider it done.